### PR TITLE
Remove OpenZeppelin

### PR DESCRIPTION
**Pull request by Vectorized**
This PR will also make updating to diamond upgradeables easier.

### CODE REVIEW

File: contracts/ERC721A.sol [hunk 1]
```diff
@@ -88,11 +92,11 @@ contract ERC721A is Context, ERC165, IERC721A {
     /**
      * @dev See {IERC165-supportsInterface}.
      */
-    function supportsInterface(bytes4 interfaceId) public view virtual override(ERC165, IERC165) returns (bool) {
+    function supportsInterface(bytes4 interfaceId) public view virtual override returns (bool) {
```
Comment (cygaar) [inline contracts/ERC721A.sol:95]: I would leave a comment in the code stating what these values are and where they're coming from

File: contracts/ERC721A.sol [hunk 2]
```diff
@@ -276,7 +280,7 @@ contract ERC721A is Context, ERC165, IERC721A {
         bytes memory _data
     ) public virtual override {
         _transfer(from, to, tokenId);
-        if (to.isContract()) if(!_checkContractOnERC721Received(from, to, tokenId, _data)) {
+        if (to.code.length != 0) if(!_checkContractOnERC721Received(from, to, tokenId, _data)) {
```
Comment (cygaar) [inline contracts/ERC721A.sol:283]: Is checking code.length equivalent to using the `EXTCODESIZE` opcode?

File: contracts/ERC721A.sol [hunk 3]
```diff
@@ -5,11 +5,18 @@
...
+ * @dev ERC721 token receiver interface.
+ */
+interface __ERC721A__IERC721Receiver {
+    function onERC721Received(
+        address operator,
+        address from,
+        uint256 tokenId,
+        bytes calldata data
+    ) external returns (bytes4);
+}
```
[the remark this answers is not in the document]
Comment (cygaar) [reply contracts/ERC721A.sol:19]: Any reason you're doing __ERC721A instead of just ERC721A?

File: contracts/ERC721A.sol [hunk 4]
```diff
@@ -596,4 +601,44 @@ contract ERC721A is Context, ERC165, IERC721A {
...
+     * @dev Returns the message data (defaults to `msg.data`).
+     */
+    function _msgData() internal view virtual returns (bytes calldata) {
+        return msg.data;
+    }
+
+    /**
+     * @dev Converts a `uint256` to its ASCII `string` decimal representation.
+     */
+    function _toString(uint256 value) internal pure returns (string memory) {
```
Comment (cygaar) [inline contracts/ERC721A.sol:622]: Might be worth throwing a test or two for this into our test suite

**Review by cygaar — COMMENTED** [review]
Small comments otherwise lgtm

File: contracts/ERC721A.sol (hunk 3 above)
[the remark this answers is not in the document]
Comment (Vectorized) [reply contracts/ERC721A.sol:19]: Ok, let’s go with `ERC721A__`.

File: contracts/ERC721A.sol (hunk 1 above)
Comment (Vectorized) [reply contracts/ERC721A.sol:95]: Ok, i’ll leave a comment with the EIP165 link.

File: contracts/ERC721A.sol (hunk 2 above)
Comment (Vectorized) [reply contracts/ERC721A.sol:283]: yes

File: contracts/ERC721A.sol (hunk 4 above)
Comment (Vectorized) [reply contracts/ERC721A.sol:622]: Already added.

**Pull request merged**